framework/include/outputs/Checkpoint.h

```diff
@@ -83,6 +82,12 @@ class Checkpoint : public FileOutput
   /// Sets the autosave flag manually if the object has already been initialized.
   void setAutosaveFlag(CheckpointType flag) { _checkpoint_type = flag; }
 
+  /// Gets information about the checkpoint
+  const std::vector<std::pair<std::string, std::string>> & getCheckpointInfo() const
```

Comment:
Few things:

- Reduce the need to record it. Just call it on the fly in the header, this reduces the need to do the set/get thing. This isn't expensive.
- Don't make it vec<pair<...>>, just do a vector of the preformatted things that you can loop to and explain in the docstring what it'll look like

Reply:
Please see my changes and resolve this conversation if you are satisfied.